packages/wallet/test/hardware/trezor/TrezorKeyAgent.test.ts

```diff
@@ -66,6 +66,196 @@ describe('TrezorKeyAgent', () => {
+      props = {
+        options: {
+          validityInterval: {
+            invalidBefore: Cardano.Slot(1),
+            invalidHereafter: Cardano.Slot(999_999_999)
+          }
+        },
+        outputs: new Set<Cardano.TxOut>([outputs.simpleOutput])
+      };
+      txInternals = await wallet.initializeTx(props);
+
+      const signatures = await keyAgent.signTransaction({
+        body: txInternals.body,
+        hash: txInternals.hash
+      });
+      expect(signatures.size).toBe(2);
+    });
+
+    it('successfully signs a transaction that mints a token', async () => {
+      const policyId = Cardano.PolicyId('38299ce86f8cbef9ebeecc2e94370cb49196d60c93797fffb71d3932');
+      const assetId = Cardano.AssetId(`${policyId}707572706C65`);
+      props = {
+        mint: new Map([[assetId, 1n]]),
+        outputs: new Set<Cardano.TxOut>([outputs.simpleOutput])
+      };
+      txInternals = await wallet.initializeTx(props);
+
+      const signatures = await keyAgent.signTransaction({
+        body: txInternals.body,
+        hash: txInternals.hash
+      });
+      expect(signatures.size).toBe(2);
+    });
+
+    it('successfully signs stake registration and delegation transaction', async () => {
+      const rewardAccount = keyAgent.knownAddresses[0].rewardAccount;
+      const stakeKeyHash = Cardano.RewardAccount.toHash(rewardAccount);
+      const stakeRegistrationCert = {
+        __typename: Cardano.CertificateType.StakeKeyRegistration,
+        stakeKeyHash
+      } as Cardano.StakeAddressCertificate;
+      const stakeDelegationCert = {
+        __typename: Cardano.CertificateType.StakeDelegation,
+        poolId,
+        stakeKeyHash
+      } as Cardano.StakeDelegationCertificate;
+
+      props = {
+        certificates: [stakeRegistrationCert, stakeDelegationCert],
+        outputs: new Set<Cardano.TxOut>([outputs.simpleOutput])
+      };
+      txInternals = await wallet.initializeTx(props);
+
+      const signatures = await keyAgent.signTransaction({
+        body: txInternals.body,
+        hash: txInternals.hash
+      });
+      expect(signatures.size).toBe(2);
+    });
+
+    it('successfully signs stake deregistration transaction', async () => {
+      const rewardAccount = keyAgent.knownAddresses[0].rewardAccount;
+      const stakeKeyHash = Cardano.RewardAccount.toHash(rewardAccount);
+      const stakeDeregistrationCert = {
+        __typename: Cardano.CertificateType.StakeKeyDeregistration,
+        stakeKeyHash
+      } as Cardano.StakeAddressCertificate;
+
+      props = {
+        certificates: [stakeDeregistrationCert],
+        outputs: new Set<Cardano.TxOut>([outputs.simpleOutput])
+      };
+      txInternals = await wallet.initializeTx(props);
+
+      const signatures = await keyAgent.signTransaction({
+        body: txInternals.body,
+        hash: txInternals.hash
+      });
+      expect(signatures.size).toBe(2);
+    });
+
+    it.skip('successfully signs pool registration transaction', async () => {
```

Comment:
is the skip intentional?

Reply:
yes, since it is not possible to sign pool registration tx with change outputs but I wanted to keep the test.

Reply:
> it is not possible to sign pool registration tx with change outputs

Is this a Trezor firmware limitation?

Reply:
@mkazlauskas not sure about FW but it is a `trezor/connect` package limitation. I also checked their tests and it is tested without change output.